### DISCUSSION TITLE
Cannot connect to Obsidian after upgrading from 2.2.4 to 3.0.0 on Brave

### DISCUSSION THREAD

**Discussion by maxox** [coddingtonbear/obsidian-web#134] [Exotic Browser Support]
I just update the extension from 2.2.4 to 3.0.0. from then the extension is unable to connect with my obsidian when i click on it (obsidian version 1.1.16 on windows 7).
do i need to downgrade it to 2.2.4 or there is gonna be a fix of it?

**Comment by maxox** [comment]
i just downgraded it now its fine

**Reply by iwyatt** [reply]
What process did you use to downgrade?

**Reply by maxox** [reply]
I downloaded the previous version of .crx file from 3rd party website, unzip the file, then use "load unpack"

**Comment by coddingtonbear** [comment]
I'm afraid I can't make any guesses without a bit more information.  I think the following would help:

1. A screenshot of the "Options" page -- you can get to that by right-clicking on the plugin icon and selecting "Options".  I recommend blurring out the API key section so you don't give others a hint about your API key.
2. Let's see what your extension logs say.  While on your "Options" page, open "Developer Tools" (`Ctrl+Shift+I`, usually), go to the "Console" tab of the developer tools window, then right-click on the background of the bit where you see logging messages printed and select "Save As". Upload the saved document to this tissue.
3. A screenshot of the Chrome-provided "Manage extension" page that you can also find from the displayed menu when right-clicking on the icon in the toolbar.  Specifically, the part I'd need to see is the "Site access" section.

**Comment by dce** [comment]
I started seeing this this morning, as well, but only in Brave (Chrome seems to work fine once I installed the plugin). In Brave, I see the following when I click the plugin button:

<img width="430" alt="image" src="https://github.com/coddingtonbear/obsidian-web/assets/3478/309b0c32-41b9-4560-a9df-b15be84042fe">

If I turn Brave's "shields" off, all works as expected.

**Comment by coddingtonbear** [comment]
I assume, then, that you, too, were using Brave, @maxox?  

My only guess is that this might be rooted in some changes to how permissions are requested/handled in the new version.  The way permissions are handled in 3.0.0 is a little different than 2.2.4 in that the earlier version's manifest declared that it needs access to 127.0.0.1 on both of the ports used, but the later version does _not_ declare any needed permissions in the manifest, and they're instead asked for when you configure the app.

Since both you and I,@dce, are not having any difficulty in Chrome, I'm going to move this into a discussion from an Issue since I'm afraid that although other browsers might have some compatibility with Chrome extensions, I don't personally guarantee compatibility with anything other than Chrome.

**Reply by maxox** [reply]
yes, im using brave. i update the extension to 3.0.1 and it works now, thanks.

**Comment by iwyatt** [comment]
I'm also having this issue. Also using Brave. 

I respect the author chooses to focus and support only the platform they prefer since they publish this for free.

Will look in to making a local copy work for my own purposes. If I make progress on that, I'll share the update here. 

I found a [bookmarklet](https://forum.obsidian.md/t/obsidian-web-clipper-bookmarklet-with-full-markdown-support-for-images-headings-and-code-blocks/22068) which might be a sufficient work-around for some people (but not my own workflows).

**Comment by dce** [comment]
This is kind of interesting. The first time I click the plugin button, I see the aforementioned error message:

<img width="1992" alt="image" src="https://github.com/coddingtonbear/obsidian-web/assets/3478/667a1cca-2abc-4dec-880b-4595aad637a0">

If I click it a few more times, I get a different error:

<img width="1992" alt="image" src="https://github.com/coddingtonbear/obsidian-web/assets/3478/4c7645f4-359a-49f3-b623-4c9c9788a1a1">

I'm with @iwyatt though -- seems like a Brave issue, not an issue specific to this plugin. Turning off shields when I want to save a page isn't terribly burdensome.

**Reply by coddingtonbear** [reply]
The second message you're seeing there is probably rooted in just bad luck when opening/closing the extension window repeatedly. For rendering templates, we communicate with a "sandboxed" iframe using some message passing logic.  If you're closing/re-opening the extension multiple times, it's totally possible for a later instantiation of the window to receive a message intended for an earlier instantiation and you'd then see such a message because your later instantiation won't know anything about the template rendering requests made by the earlier instantiation.

**Comment by Mikle-Bond** [comment]
I'm not sure I have the exact same problem, but it appeared at the same time, so it might be related. I'm facing CORS issues.

The errors also remain on https://coddingtonbear.github.io/obsidian-local-rest-api. Looking at network tab reveals that it is probably the same issue after all, as the same preflight responses block requests. Trying with `curl` succeeds. 

```
Access to fetch at 'http://127.0.0.1:27123/vault/' from origin 'https://coddingtonbear.github.io' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Private-Network' header was present in the preflight response for this private network request targeting the `local` address space.
```

I'm using Cent Browser `Version 5.0.1002.354 (Official Build) (64-bit) (Chromium 102.0.5005.167)`

**Reply by Mikle-Bond** [reply]
My investigation yields some results. I've setup a mitm proxy using `caddy` and configured it to inject the `access-control-allow-private-network: true` in every response. And now extension works.

I wanted to compare with the v2.2.4, but when I rolled back, I couldn't see any messages in console and no requests in "network" tab. This leads me to the idea of where the problem could've have appeared.

https://github.com/coddingtonbear/obsidian-web/compare/2.2.4...3.0.0#diff-9a8dfe07c2bfeb25a7bcb4e293c44857c0bbbfc3417c31f1dc9d1d3482864216R109-R120

Extension uses `chrome.scripting` API to run `popup.js` inside the page, instead of running it inside the `popup.html`. This means, that previously all the request were made from the context of extension itself and didn't trigger any CORS requests (or they were somehow more permissive due to `127.0.0.1` mentioned in `host_permissions` in manifest. Now the script runs on behalf of the page and requires `access-control-allow-private-network` header.

 @coddingtonbear, maybe this header should be added to Local REST API plugin? I'm not sure how could this even work without it...

**Reply by Mikle-Bond** [reply]
For reference, here's the issue about CORS and private networks for ExpressJS: https://github.com/expressjs/cors/issues/236

There's also a workaround mentioned in the comments.

**Reply by coddingtonbear** [reply]
I did a little reading and some experimentation, and I can say that we _shouldn't_ need to worry about CORS since we're requesting permissions (see https://developer.chrome.com/docs/extensions/mv3/network-requests/#requesting-permission).  We're are using `optional_host_permissions`, so that means it's not declared in the manifest and is instead requested (via a prompt) when you select a hostname, but as far as Chrome is concerned, that's not important.

That said, in my experimentation in Chrome, I did discover that there is a bug when attempting to connect to a host other than `127.0.0.1`.  Requests _from the popup_ (the options page itself is fine) will fail to be made if you select any hostname other than `127.0.0.1`.  This is because we're making those requests from the page's security origin instead of that of the extension.  Whoops.  I'll enter a bug for fixing that.  I have to say, though, that I'm not remotely confident that fixing that will have any impact on the problems Brave users are experiencing.

Posted as #137.

**Reply by iwyatt** [reply]
Thanks for the updates, both. @Mikle-Bond has made more progress than me in less, time, so I switched to looking for work-arounds. 
I have a suspicion and intuition that the bug @coddingtonbear is working on is tangentially related. If I were more knowledgeable, I'd pull on that bug's metaphorical-thread a bit to see if/how it affects Brave and other chromium browsers.

# Work-Around
Although it is possible to disable ad blocking in Brave in order to get the extension to work (per @dce ), I did find another work-around:  I noticed that Brave was ("correctly") blocking `127.0.0.1` in its list of blocked ads and trackers when on any particular site. Since it is considered an ad/tracker, I figured I should be able to white list it in Brave's ad blocking rules engine. Which led me to the following: 

 ***
## Whitelisting a specific Tracker : r/brave_browser
URL: https://www.reddit.com/r/brave_browser/comments/ze3jgj/whitelisting_a_specific_tracker/

> The way to whitelist it is to basically go to the custom filters in brave://adblock and enter
> 
> @@*tracker name*
> 
> save, restart the browser and it will work !
> 
> Edit : Refer to [this](https://help.adblockplus.org/hc/en-us/articles/360062733293) link for more information !

***

The above worked for me without having to exit and restart the browser. I did note however that an https site will display `not secure`  in the address bar after activating the plug-in. My snippet successfully saved to my obsidian vault.

**Reply by dce** [reply]
Yeah, this works for me as well:

<img width="675" alt="image" src="https://github.com/coddingtonbear/obsidian-web/assets/3478/4068d73b-7576-4d50-a52c-9959a3abc0f9">

Thanks @iwyatt

**Reply by coddingtonbear** [reply]
I pulled together a release earlier today for https://github.com/coddingtonbear/obsidian-web/discussions/134#discussioncomment-7567736 -- so I'm curious to see if that new version fixes your problems, too.  Please keep in mind that it'll take time for your browser to notice that the update is available.  You can tell which version you have installed by right clicking on the extension icon, clicking the "Manage extension" button, and seeing what value is listed for "Version".  Version 3.0.1 has the fix for my issue above and might accidentally have the fix for yours, too.

**Reply by Mikle-Bond** [reply]
@coddingtonbear Yay, extension works. Thank you for quick response!

On the downside I want to note that in my case, before the fix, the extension didn't want to communicate with `127.0.0.1` either. And this solution still does not solve the problem with https://coddingtonbear.github.io/obsidian-local-rest-api. I wonder, am I the only one who can't communicate with obsidian from the Swagger instance? It works fine if it's served on `localhost`, but public instance is broken for me.

I found that there's a way to disable CORS for private network from `chrome://flags` menu. Options are:

- Block insecure private network requests
- Send Private Network Access preflights
- Respect the result of Private Network Access preflights

Tweaking all three makes it work again. But I would prefer not to disable security features)

**Reply by coddingtonbear** [reply]
No -- I wouldn't expect that to be able to reach non-localhost hostnames there since Obsidian Local Rest API doesn't implement CORS.  That's part of why it's not configurable there.

**Reply by iwyatt** [reply]
@dce - did the `3.0.1` version that got pushed today break your extension (given the workaround we used to get `3.0.0` to work)?

Mine stopped working again and I don't know if has to do with `3.0.1` or something else.

**Reply by iwyatt** [reply]
** **Sorry for the rambling diagnosis... will clean this up tomorrow morning to be more clear** **

Interestingly, when I look at the network properties, I do see the failed requests which indicate an expired certificate. I'm on my windows machine at the moment, and when I re-download the cert from local rest api plugin to install it, it shows dates from October.
![image](https://github.com/coddingtonbear/obsidian-web/assets/7514713/1b968785-99fc-46ba-8f6b-5fe416e79e83)
![image](https://github.com/coddingtonbear/obsidian-web/assets/7514713/c5621860-212d-4acf-afef-0e61f56c0fc8)

I'm trying to `reset crypo[sic]` in the local rest api plugin to generate a new certificate to see if that helps... 

Regenerating it and re-downloading the cert did give me one with fresh date ranges...
![image](https://github.com/coddingtonbear/obsidian-web/assets/7514713/9069d792-f7b7-4231-9d02-fba9fd4e6b62)

However, now I'm getting an error that the certificate authority is invalid...
![image](https://github.com/coddingtonbear/obsidian-web/assets/7514713/dd57bd46-4523-457c-ad79-7030ed743be2)

I tried importing into a different location based on what Window's recommended, completely exited Brave, and then started it up again, and as of the moment, the plugin works again - but only with Brave's shields turned off. 

I do see that Brave's shield for 'block scripts' seems to be affecting the extension as well, even though Brave only shows scripts specific to the page, not the extension. 
![image](https://github.com/coddingtonbear/obsidian-web/assets/7514713/6eda5537-be52-47c2-ae75-79aad26822ba)

When I look at the logs for the extension's service worker in Brave, and try with scripts blocked and then not-blocked, I see the key difference appears to be that with Shields On, that HTTP POST gets a `404 not found` status, and the headers don't get populated for `Content-Location`. The `Request URL` while the scripts are blocked is also truncated: `https://127.0.0.1:27124/` instead of `https://127.0.0.1:27124/periodic/daily/` .

So it seems a script in the extension is getting blocked that populates the `Content-Location` header  on the HTTP POST to insert the content into the daily note, which results in a 404, and the content doesn't get sent to Obsidian's Local Rest API.  Interestingly, HTTP GET does not get blocked, and the HTTP POSTs that do that call `search` api do not get blocked either. 

![image](https://github.com/coddingtonbear/obsidian-web/assets/7514713/c310846b-a46c-4ab5-9d09-2b59dcf4859d)

Looking at `background.js` I see: 
```javascript
            function h(e, t, n, o, a) {
                var i;
                return r(this, void 0, void 0, (function*() {
                    const r = Object.assign(Object.assign({}, o), {
                        headers: Object.assign(Object.assign({}, o.headers), {
                            Authorization: `Bearer ${t}`
                        }),
                        method: null === (i = o.method) || void 0 === i ? void 0 : i.toUpperCase(),
                        mode: "cors"
                    });
                    return fetch(`http ${a ? "" : "s"}://${e}:${a ? "27123" : "27124"}${n}`, r)
                }
                ))
            }
```

Note that `mode: "cors"` there. Does this mode cause issues since Obsidian Local Rest API doesn't implement it?

**Reply by dce** [reply]
I can replicate this -- works w/ shields down, w/ shields up I get:

<img width="400" alt="image" src="https://github.com/coddingtonbear/obsidian-web/assets/3478/ff515698-2c7b-41a8-984c-3989078d9d19">

**Reply by coddingtonbear** [reply]
I can't lend a lot of insight into how Brave does anything, and really can't offer any meaningful support for this, but I can say that that sounds like maybe Brave is making the request execute outside of the script's security origin for some reason.  Requests are only allowed to define certain parameters (see "Simple Requests" here: https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS#simple_requests) when made within the page security origin unless the target API implements CORS, and Obsidian Local REST API doesn't.

My only guess for you all: I'd be suspicious that the extension isn't granted permission to the relevant host.  When you enter a hostname to connect to, Obsidian Web will prompt you for permission to connect to that host -- maybe that "Request permissions" dialog is blocked on Brave?  I don't know what the extension settings panel looks like on Brave, but on Chrome you see this: 

![image](https://github.com/coddingtonbear/obsidian-web/assets/527661/66394dcf-456b-4d4c-84b6-34a3c9fbd66d)

The important bit is that "Site Access" section.  It should show your selected hosts on both of the ports you see shown there.

**Reply by Mikle-Bond** [reply]
@coddingtonbear 

> unless the target API implements CORS, and Obsidian Local REST API doesn't.

WDYM it doesn't implement it? I'm looking at router setup routine and see it assign the CORS middleware: https://github.com/coddingtonbear/obsidian-local-rest-api/blob/08df581497fa6a724904a2850524e0f1d8a4a9dc/src/requestHandler.ts#L991

And I check with `curl` to see the preflight responses:

```
curl -vks -X OPTIONS https://127.0.0.1:27124/vault/
...
< HTTP/1.1 204 No Content
< X-Powered-By: Express
< Access-Control-Allow-Origin: *
< Access-Control-Allow-Methods: GET,HEAD,PUT,PATCH,POST,DELETE
< Vary: Access-Control-Request-Headers
< Content-Length: 0
< X-Response-Time: 0.711ms
< Date: Sun, 19 Nov 2023 07:04:23 GMT
< Connection: keep-alive
< Keep-Alive: timeout=5
<
```

Seems aligned with the default settings for the middleware. The only thing is that it ignores "access-control-request-private-network" request and thus makes Chrome angry.

**Reply by coddingtonbear** [reply]
Oh, neat, I guess I did implement that there!

**Comment by daniel347x** [comment]
I am having exactly the same problem and I am on Chrome.

Super sorry I don't have time to post details - maybe soon if I can't resolve the issue.

I just want to indicate that this was working for me a few days ago, but now I'm getting the same error as https://github.com/coddingtonbear/obsidian-web/discussions/134#discussioncomment-7588879 and others in this thread...

Another message:

![image](https://github.com/coddingtonbear/obsidian-web/assets/309746/6fe828f9-9878-4572-bf31-4496244a374b)

(And, connection does succeed from https://coddingtonbear.github.io/obsidian-local-rest-api - it only fails from my normal tabs.)

**Reply by coddingtonbear** [reply]
@daniel347x that particular message is not discussed in this thread.  I'd recommend creating a separate bug report, but I have to say that it's working for me and others on Chrome, so whatever is going wrong is probably related to the particulars of your environment.

**Comment by lduperval** [comment]
Hate to say it but... me too.

Has anyone found a way to fix this? If not, how do I downgrade to a pre-3.0 version? Someone mentioned going to another website, but I don't know which one.

@coddingtonbear would you consider putting a 2.4 version with a different name in the webstore?

**Reply by maxox** [reply]
here you are. crx4chrome seems the only one available.

**Reply by coddingtonbear** [reply]
I'm afraid I won't be making a Brave-specific build, no, but there are [instructions](https://github.com/coddingtonbear/obsidian-web/wiki/Development) for how to do a development build of your own.  It should just be as easy as doing those documented steps while having the 2.2.4 tag checked out.

Also: I welcome folks using that exact same documented process for trying to figure out what the problem is in Brave.  I'm afraid I won't be making the time to troubleshoot in browsers other than Chrome (and it works just fine there!), but if some of you are familiar with Brave, it probably won't be too hard to narrow-in on what's going wrong.  What I'm suspicious of is that maybe Brave doesn't follow Chrome's permission requesting/granting behavior, but can only guess.

**Comment by charles-vdulac** [comment]
I'm not sure I have the same problem, but it appeared simultaneously, so it might be related. 

Extension Version 3.0.1
Brave: latest version to date.

I was facing this issue: 
<img width="954" alt="Screenshot 2024-01-01 at 15 30 58" src="https://github.com/coddingtonbear/obsidian-web/assets/783168/2403a7a2-1041-40ea-aa71-c8c58e4b570c">

<img width="1355" alt="Screenshot 2024-01-01 at 15 31 35" src="https://github.com/coddingtonbear/obsidian-web/assets/783168/ef33d723-0a14-4038-8167-2ee390024fb6">

Note: It is not clear to me why the list view displays `NET::ERR_CERT_AUTHORITY_INVALID`, but the details view displays `strict-origin-when-cross-origin`. Anyway, that doesn't matter.

**What I did to fix it:**

I'm on Mac OS.

* I downloaded the local server certificate file from the "Local Rest API" setting page.
* I installed it: `sudo security add-trusted-cert -k /Library/Keychains/System.keychain /tmp/obsidian-local-rest-api.crt`
* I opened KeyChain. The `Obsidian Local REST API` entry was in red as the root certificate authority was not trusted. 
* I selected it, and in the trust section, I selected the option `Always Trust` when using this certificate.

**Reply by coddingtonbear** [reply]
Oh, interesting!  My understanding is that this extension just _doesn't_ work on Brave at all anymore, so if you've found a way to get it working at all, I think you've may have found the problem entirely.

Unfortunately, of course, every platform (linux, mac, windows) has its own process for adding/trusting root certificates; so I can't say if the instructions you've given above would solve it for everybody, but I bet it'll give folks an idea of what to do at least.  Good work on finding a potential lead, @charles-vdulac!

**Comment by lduperval** [comment]
Hmm, I have a Windows machine and can't do the sudo magic. But I found out that the certificate I was importing didn't seem to be recognized by Brave. It seemed to be stored in the wrong place. So instead of importing the certificate as a regular user from Brave, I used the Certificate manager as admin and imported the obsidian certificat there.

Now, Brave says it can connect securely to the client. Yay!

But I still get a failure. Boo!

But the failure seems to be in the app... I think. Booyay??

Here it is while trying to add the site at https://minigpt-4.github.io/:

```
 background.js:2 Obsidian request failed TypeError: Failed to fetch
    at background.js:2:13242
    at Generator.next (<anonymous>)
    at background.js:2:11680
    at new Promise (<anonymous>)
    at background.js:2:11425
    at h (background.js:2:13001)
    at background.js:2:9564
(anonymous) @ background.js:2
Promise.catch (async)
(anonymous) @ background.js:2
Promise.then (async)
(anonymous) @ background.js:2
VM71 vendor.js:205 Uncaught Error: Received template compilation callback, but could not identify message: {"type":"response","success":true,"request":{"command":"render","id":"f83a412c-c1a3-4534-b183-dc81997dec44","template":"## {{page.title}}\n\nURL: {{page.url}}\n\n\n{{#if page.selectedText}}\n\n{{quote page.selectedText}}\n{{/if}}","context":{"page":{"url":"https://minigpt-4.github.io/","title":"Minigpt-4","selectedText":"","content":"## MiniGPT-4:\n\n## Enhancing Vision-language Understanding with Advanced Large Language Models\n\n**▶** King Abdullah University of Science and Technology\n\n\\*Equal Contribution\n\n  \n\n## Abstract\n\nThe recent GPT-4 has demonstrated extraordinary multi-modal abilities, such as directly generating websites from handwritten text and identifying humorous elements within images. These features are rarely observed in previous vision-language models. We believe the primary reason for GPT-4's advanced multi-modal generation capabilities lies in the utilization of a more advanced large language model (LLM). To examine this phenomenon, we present MiniGPT-4, which aligns a frozen visual encoder with a frozen LLM, Vicuna, using just one projection layer. Our findings reveal that MiniGPT-4 possesses many capabilities similar to those exhibited by GPT-4 like detailed image description generation and website creation from hand-written drafts. Furthermore, we also observe other emerging capabilities in MiniGPT-4, including writing stories and poems inspired by given images, providing solutions to problems shown in images, teaching users how to cook based on food photos, etc. In our experiment, we found that only performing the pretraining on raw image-text pairs could produce unnatural language outputs that lack coherency including repetition and fragmented sentences. To address this problem, we curate a high-quality, well-aligned dataset in the second stage to finetune our model using a conversational template. This step proved crucial for augmenting the model's generation reliability and overall usability. Notably, our model is highly computationally efficient, as we only train a projection layer utilizing approximately 5 million aligned image-text pairs.\n\n## Video Presentation\n\n## Model\n\n**MiniGPT-4 consists of a vision encoder with a pretrained ViT and Q-Former, a single linear projection layer, and an advanced Vicuna large language model. MiniGPT-4 only requires training the linear layer to align the visual features with the Vicuna.**:\n\n![](https://minigpt-4.github.io/images/overview.png)\n\n### \n\n**The architecture of MiniGPT-4.**\n\n## BibTeX\n\n```\n\n\n@article{zhu2023minigpt,\n  title={MiniGPT-4: Enhancing Vision-Language Understanding with Advanced Large Language Models},\n  author={Zhu, Deyao and Chen, Jun and Shen, Xiaoqian and Li, Xiang and Elhoseiny, Mohamed},\n  journal={arXiv preprint arXiv:2304.10592},\n  year={2023}\n}\n```\n\n## Results"},"article":{"title":"","length":2979,"excerpt":"▶  King Abdullah University of Science and Technology","byline":"Deyao Zhu*\n                ,","dir":null,"siteName":null}}},"rendered":"## Minigpt-4\n\nURL: https://minigpt-4.github.io/\n\n\n"}
    at VM71 vendor.js:205:661392
(anonymous) @ VM71 vendor.js:205
```

I tried another page and got this (https://arxiv.org/abs/2311.07590):

```
background.js:2 Obsidian request failed TypeError: Failed to fetch
    at background.js:2:13242
    at Generator.next (<anonymous>)
    at background.js:2:11680
    at new Promise (<anonymous>)
    at background.js:2:11425
    at h (background.js:2:13001)
    at background.js:2:9564
(anonymous) @ background.js:2
Promise.catch (async)
(anonymous) @ background.js:2
Promise.then (async)
(anonymous) @ background.js:2
background.js:2 TypeError: Failed to fetch
    at background.js:2:13242
    at Generator.next (<anonymous>)
    at background.js:2:11680
    at new Promise (<anonymous>)
    at background.js:2:11425
    at h (background.js:2:13001)
    at background.js:2:12688
    at Generator.next (<anonymous>)
    at background.js:2:11680
    at new Promise (<anonymous>)
(anonymous) @ background.js:2
s @ background.js:2
Promise.then (async)
c @ background.js:2
i @ background.js:2
Promise.then (async)
c @ background.js:2
(anonymous) @ background.js:2
(anonymous) @ background.js:2
(anonymous) @ background.js:2
background.js:2 TypeError: Failed to fetch
    at background.js:2:13242
    at Generator.next (<anonymous>)
    at background.js:2:11680
    at new Promise (<anonymous>)
    at background.js:2:11425
    at h (background.js:2:13001)
    at background.js:2:12688
    at Generator.next (<anonymous>)
    at background.js:2:11680
    at new Promise (<anonymous>)
(anonymous) @ background.js:2
s @ background.js:2
Promise.then (async)
c @ background.js:2
i @ background.js:2
Promise.then (async)
c @ background.js:2
(anonymous) @ background.js:2
(anonymous) @ background.js:2
(anonymous) @ background.js:2
```

**Reply by coddingtonbear** [reply]
I'm sure you've came to this conclusion already, but that message -- "failed to fetch" -- almost certainly means that the browser rejected our request to talk to Obsidian.   It's difficult to say what about the situation on Brave makes the browser reject these requests, but when this error occurs on Chrome it could occur for a certificate, CORS, or host permissions problem so I'm suspicious that Brave might not be either prompting or allowing for the host-level permissions that we ask for as part of set-up.

I bet that somebody who had both Brave and Chrome installed might be able to figure out what's different by just opening up Chrome and Brave side-by-side, and in each window installing the extension and walking through the configuration process.  I'd _guess_ that you'll find a difference and my guess would be that in Chrome you see a permissions request dialog at one step, and you won't in Brave.

**Comment by lduperval** [comment]
And another message I saw for the first time:

![image](https://github.com/coddingtonbear/obsidian-web/assets/73903942/8b9bf441-988d-40d8-b8c5-95c83df5caae)

in case it can help others who are working on it.

**Comment by neutron246** [comment]
I am not sure if this is still an actual issue. I have Brave too and had problems making it work. But the issue ended up being with the certificate, not the Browser. Even though it had a green checkmark next to the API Key, it only worked after I imported the certificate into "Trusted Root Certificate Authorities". As it was solved in #155.

**Comment by lduperval** [comment]
Wow, that worked, thank you!

I'm not sure what you meant when you said "I reset the Crypto, and then had to re-import the certificate into Chrome" in comment #155. This is what I did.

On Windows, I went to the certificate manager tool. I found the Local Rest API certificate (it was under personal) and I dragged and dropped it to the Trusted Root Certificate Authorities folder. Then I tested the official plugin and it worked.

Thank you very much!
